tck-test/BaseSubscriber.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <folly/ExceptionWrapper.h>
+#include <condition_variable>
+#include <mutex>
+#include <vector>
+#include "src/Payload.h"
+#include "yarpl/Refcounted.h"
+
+namespace rsocket {
+namespace tck {
+
+class BaseSubscriber : public virtual yarpl::Refcounted {
+ public:
+  virtual void request(int n) = 0;
+  virtual void cancel() = 0;
+  virtual void awaitTerminalEvent();
+  virtual void awaitAtLeast(int numItems);
+  virtual void awaitNoEvents(int waitTime);
+  virtual void assertNoErrors();
+  virtual void assertError();
```

Comment:
nit: is there value in checking if the error/exception is matching with the expected one. Like we expect one kind of exception and we get that kind of exception, not just we just got an error.